Modules/itertoolsmodule.c

```diff
@@ -990,7 +955,7 @@ tee_copy(teeobject *to, PyObject *Py_UNUSED(ignored))
 {
     teeobject *newto;
 
-    newto = PyObject_GC_New(teeobject, &tee_type);
+    newto = PyObject_GC_New(teeobject, Py_TYPE(to));
```

Comment:
We could use `to->state->tee_type` here, but `Py_TYPE(to)` is one pointer chase less.